api/types/access_request_test.go

```diff
@@ -18,12 +18,52 @@ package types
 
 import (
 	"testing"
+	"time"
 
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
 	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/constants"
 )
 
 func TestAssertAccessRequestImplementsResourceWithLabels(t *testing.T) {
 	ar, err := NewAccessRequest("test", "test", "test")
 	require.NoError(t, err)
 	require.Implements(t, (*ResourceWithLabels)(nil), ar)
 }
+
+func TestValidateAssumeStartTime(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	creation := clock.Now().UTC()
+	day := 24 * time.Hour
+
+	expiry := creation.Add(12 * day)
+	maxAssumeStartDuration := creation.Add(constants.MaxAssumeStartDuration)
+
+	// Start time too far in the future.
+	invalidMaxedAssumeStartTime := creation.Add(constants.MaxAssumeStartDuration + (1 * day))
+	err := ValidateAssumeStartTime(invalidMaxedAssumeStartTime, expiry, creation)
+	require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
```

Comment:
There are a lot of other test cases within this test that you can still run even though this one failed. This would create a faster feedback loop.

https://go.dev/wiki/TestComments#keep-going